### PR TITLE
fix: polyfill Object.hasOwn for node 14

### PR DESCRIPTION
**Pull request by ljharb**
From #4060; fixes https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/953.

### CODE REVIEW

**Review by straker — DISMISSED** [review]
Thanks for taking on fixing < node 14.

It looks like [core-js provides an implementation for Object.hasOwn](https://github.com/zloirock/core-js/blob/master/packages/core-js/actual/object/has-own.js), which we already depend on. We should use that one rather than bring in another dependency.

Also, the way we handle doing these types of polyfills in axe-core is to add them to the [imports file](https://github.com/dequelabs/axe-core/blob/develop/lib/core/imports/index.js) and only apply the polyfill if the current implementation does not exist. That way we don't have to rewrite every use of the function and can easily remove it when we don't need the polyfill anymore.

```js
import hasOwn from 'core-js-pure/actual/object/has-own';
 
if (!('hasOwn' in Object) {
  Object.hasOwn = hasOwn;
}
```

**Comment by ljharb** [comment]
Sounds good, done. It seems like tests aren't running in every supported version of node, which would have caught this as well as #4127. I'm not familiar enough with circle to fix that myself, but it'd be great if someone was able to do that separately :-)

**Comment by straker** [comment]
Yep, was actually looking into that myself to see what it would entail.

File: lib/core/imports/index.js [hunk 1]
```diff
@@ -7,6 +7,11 @@ import Color from 'colorjs.io';
 import es6promise from 'es6-promise';
 import { Uint32Array } from 'typedarray';
 import 'weakmap-polyfill';
+import hasOwn from 'core-js-pure/actual/object/has-own';
+
+if (!('hasOwn' in Object) {
```
Comment (straker) [inline lib/core/imports/index.js:12]: My bad, seems I forgot the end bracket :D

```suggestion
if (!('hasOwn' in Object)) {
```

Comment (ljharb) [reply lib/core/imports/index.js]: serves me right for naively copypasting

**Review by dbjorge — APPROVED** [review]

**Comment by straker** [comment]
Approved for security

**Comment by ljharb** [comment]
Thanks! I ran `npm run fmt` locally and no diff was generated, so i'm not sure why the prettier task was failing.

**Comment by ljharb** [comment]
Hopefully this will make it into v4.8.2?

**Comment by straker** [comment]
Yep, I forgot I wanted this into the 4.8.1 release before we made it.

**Comment by ljharb** [comment]
Thank you!

**Pull request merged**